update stats and plots driver/scripts

BinbinZhou-NOAA opened the pull request:
## Pull Request Testing ##

- [ y] Describe testing already performed for this Pull Request:</br>

- [ y] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- [ y] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [y ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [y ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [y ] Review the source issue metadata (required labels, projects, and milestone).
- [ y] Complete the PR description above.
- [y ] Ensure the PR title matches the feature branch name.
- [y ] Check the following:
- [y ]  Instructions provided on how to run
- [y ]  Developer's name is replaced by ${user} where necessary throughout the code
- [y ]  Check that the ecf file has all the proper definitions of variables
- [y ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [y ]  Check to see that the output directory structure is followed
- [y ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [y ] After submitting the PR, select **Development** issue with the original issue number.
- [y ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [y ] Close the linked issue.

BinbinZhou-NOAA committed:
update scripts

BinbinZhou-NOAA committed:
change the spatial map file name to match naming standard

BinbinZhou-NOAA committed:
relocate COMROOT definition in spatial map job

malloryprow commented:
@BinbinZhou-NOAA I was looking over this since it touches jobs/mesoscale/stats/JEVS_MESOSCALE_STATS.

I feel some of the changes you made are going to break things for the NAM and RAP jobs.

First the removal of

> export cycle=t${cyc}z
>  setpdy.sh
>  . $DATA/PDY

This should be put back because it is an important part of every J-JOB.

Second you removed portions of the code under _if [ $MODELNAME = nam ] || [ $MODELNAME = rap ];_ then. I'm not sure why this was done.

malloryprow commented:
@BinbinZhou-NOAA It would also be good to put back the job header information back that you removed so we can keep track of all the scripts that are using jobs/mesoscale/stats/JEVS_MESOSCALE_STATS.

BinbinZhou-NOAA committed:
Update jobs/mesoscale/stats/JEVS_MESOSCALE_STATS

BinbinZhou-NOAA committed:
Update jobs/mesoscale/stats/JEVS_MESOSCALE_STATS again

malloryprow commented:
Thank you!

BinbinZhou-NOAA commented:
Mallory,

  I have updated the JEVS_MESOSCALE_STATS file.
I have still no idea why some portion under under if [ $MODELNAME = nam ]
|| [ $MODELNAME = rap ]; then . But I remember when I tried to synced the
sref.ve fork, several previous commits were discarded
(have to do so before sync).

Binbin

On Wed, Jul 12, 2023 at 2:59 PM Mallory Row ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> It would also be
> good to put back the job header information back that you removed so we can
> keep track of all the scripts that are using
> jobs/mesoscale/stats/JEVS_MESOSCALE_STATS.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#issuecomment-1633053133>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFIIAHOCGI3ZQ4WDKBDXP3XYFANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
Hi @BinbinZhou-NOAA next time you see that message to discard commits to do a sync fork, let me or @PerryShafran-NOAA know. You definitely do not have to discard commits. It just means there are conflicts that need to be manually addressed before a sync can be completed. We definitely do not want you to have to discard commits and lose time/effort/work. One of us can teach you how to resolve the conflicts manually.

BinbinZhou-NOAA commented:
Sure,  I remember you have mentioned this before.

Thanks!

Binbin

On Wed, Jul 12, 2023 at 3:29 PM Shelley Melchior ***@***.***>
wrote:

> Hi @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> next time you
> see that message to discard commits to do a sync fork, let me or
> @PerryShafran-NOAA <https://github.com/PerryShafran-NOAA> know. You
> definitely do not have to discard commits. It just means there are
> conflicts that need to be manually addressed before a sync can be
> completed. We definitely do not want you to have to discard commits and
> lose time/effort/work. One of use can teach you how to resolve the
> conflicts manually.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#issuecomment-1633093913>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFOOGXR4SBBP7KBCKZDXP33J3ANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
update scripts/mesoscale/plots/exevs_mesoscale_atmos_sref_grid2obs_plots.sh

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref.ve

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref.ve

BinbinZhou-NOAA committed:
update machine.conf

BinbinZhou-NOAA committed:
remove line evs_ver=1.0 in all driver scripts

ShelleyMelchior-NOAA commented on modulefiles/mesoscale/mesoscale_plots.sh:
@BinbinZhou-NOAA -- Please go back and review your `modulefiles/mesoscale/mesoscale_plots.sh` file. Make sure it includes everything that is in the auth EVS develop `modulefiles/mesoscale/mesoscale_plots.sh`. I see discrepancies right now.
I would also modify line 2 to read something like this, to be all inclusive:
`# modulefile for EVS mesoscale (NAM, RAP, SREF) component, plots step`

ShelleyMelchior-NOAA commented on modulefiles/mesoscale/mesoscale_stats.sh:
@BinbinZhou-NOAA -- Please go back and review your `modulefiles/mesoscale/mesoscale_stats.sh` file. Make sure it includes everything that is in the auth EVS develop `modulefiles/mesoscale/mesoscale_stats.sh`. I see discrepancies right now.
I would also modify line 2 to read something like this, to be all inclusive:
`# modulefile for EVS mesoscale (NAM, RAP, SREF) component, stats step`

ShelleyMelchior-NOAA commented:
@BinbinZhou-NOAA  -- I am starting to take a closer look at this PR. I have provided a few comments so far to be addressed.

BinbinZhou-NOAA committed:
update mesoscale_plots.sh

ShelleyMelchior-NOAA commented on jobs/mesoscale/plots/JEVS_MESOSCALE_PLOTS:
@BinbinZhou-NOAA  -- take some care here w/ `jobs/mesoscale/plots/JEVS_MESOSCALE_PLOTS`. Please compare this job script w/ the auth EVS develop to  make sure all key features are included for all aspects of the  mesoscale component. I see a lot of differences that may not be intentional.

BinbinZhou-NOAA commented:
Shelley,

  Sure I have changed the file /modulefiles/mesoscale/mesoscale_plots.sh to
match the version in the authorized repo.
But added one extra loading module (line 30):
*module load imagemagick/${imagemagick_ver}*
This module is for converting the spatial map plot from png to gif format
required by Mallory.
Line 2 was also updated.
The new file has been uploaded to github feature/sref.ve

Binbin




On Fri, Jul 21, 2023 at 3:34 PM Shelley Melchior ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On modulefiles/mesoscale/mesoscale_stats.sh
> <https://github.com/NOAA-EMC/EVS/pull/153#discussion_r1271016521>:
>
> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> -- Please go back
> and review your modulefiles/mesoscale/mesoscale_stats.sh file. Make sure
> it includes everything that is in the auth EVS develop
> modulefiles/mesoscale/mesoscale_stats.sh. I see discrepancies right now.
> I would also modify line 2 to read something like this, to be all
> inclusive:
> # modulefile for EVS mesoscale (NAM, RAP, SREF) component, stats step
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#pullrequestreview-1541747466>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFNPIASZ34UMJMTQN2LXRLKVNANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented on scripts/mesoscale/plots/exevs_mesoscale_atmos_sref_precip_spatial_plots.sh:
```diff
@@ -69,16 +68,17 @@ python $USHevs/mesoscale/ush_sref_plot_precip_py/sref_atmos_plots.py
+
+
+tar -cvf evs.plots.sref.precip.spatial.map.v${VDATE}.tar *.gif
 
```
@BinbinZhou-NOAA I just wanted to check on this file.
I wanted to make sure the lines 71-78 are meant to be commented out?
I wanted to make sure line 81 is meant to be changed from png to gif?

ShelleyMelchior-NOAA replied:
Ah ok! Thanks @BinbinZhou-NOAA  for explaining the png to gif change in your previous comment!

ShelleyMelchior-NOAA commented:
In light of Perry's recent email message (subject: Changes needed to your filenames), you will need to update all of your sref driver scripts in stats and plots to include "mesoscale".
For example, `jevs_sref_grid2obs_past90days_plots.sh` changes to `jevs_mesoscale_sref_grid2obs_past90days_plots.sh`.
You will also need to update the job card to include "mesoscale" and spell out the verif_case.
For example, `#PBS -N jevs_sref_g2o_stats` changes to `#PBS -N jevs_mesoscale_sref_grid2obs_stats`.
This PR is a good opportunity to take care of that maintenance.

BinbinZhou-NOAA commented:
yes, line 71-87 were commented out. Line 81 is tar all gif files
Sure, I'll change the driver file names from sref to mesoscale_sref

Thanks!


On Fri, Jul 21, 2023 at 4:33 PM Shelley Melchior ***@***.***>
wrote:

> In light of Perry's recent email message (subject: Changes needed to your
> filenames), you will need to update all of your sref driver scripts in
> stats and plots to include "mesoscale".
> For example, jevs_sref_grid2obs_past90days_plots.sh changes to
> jevs_mesoscale_sref_grid2obs_past90days_plots.sh.
> You will also need to update the job card to include "mesoscale".
> For example, #PBS -N jevs_sref_g2o_stats changes to #PBS -N
> jevs_mesoscale_sref_g2o_stats.
> This PR is a good opportunity to take care of that maintenance.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#issuecomment-1646210003>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFMIHXSZVEHONOZ7F23XRLRRVANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA commented:
Shelley,

  The  jobs/mesoscale/plots/JEVS_MESOSCALE_PLOTS file needs to change to
match the authorized version.
A major difference is the definition of $COMIN and $COMOUT. If matching the
authorized version,
several paths in the plotting scripts should be changed. I'll do this work
next week.

Thanks!

Binbin


On Fri, Jul 21, 2023 at 4:02 PM Shelley Melchior ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On jobs/mesoscale/plots/JEVS_MESOSCALE_PLOTS
> <https://github.com/NOAA-EMC/EVS/pull/153#discussion_r1271045100>:
>
> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> -- take some care
> here w/ jobs/mesoscale/plots/JEVS_MESOSCALE_PLOTS. Please compare this
> job script w/ the auth EVS develop to make sure all key features are
> included for all aspects of the mesoscale component. I see a lot of
> differences that may not be intentional.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#pullrequestreview-1541778778>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFKN7Z3JFW2T7LZ4WNDXRLN6JANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
And as you are doing in the NARRE PR, remove this line from your driver scripts:
`export MET_bin_exec=bin`

BinbinZhou-NOAA commented:
Shelley,

  A question for the JEVS_MESOSCALE_PLOTS file in the authorized repo:
https://github.com/NOAA-EMC/EVS/blob/develop/jobs/mesoscale/plots/JEVS_MESOSCALE_PLOTS
line 95:
export COMOUT=${COMOUT:-$(compath.py -o $NET/$evs_ver/$STEP/$COMPONENT/$RUN
and line 99:
mkdir -p $COMOUT

Since RUN=atmos,  this means the sub-directory  /atmos  will be created
under $COMPONENT(mesoscale)?
I think sub-directory /atmos.$VDATE should be created instead of just /atmos

Binbin



On Fri, Jul 21, 2023 at 4:33 PM Shelley Melchior ***@***.***>
wrote:

> In light of Perry's recent email message (subject: Changes needed to your
> filenames), you will need to update all of your sref driver scripts in
> stats and plots to include "mesoscale".
> For example, jevs_sref_grid2obs_past90days_plots.sh changes to
> jevs_mesoscale_sref_grid2obs_past90days_plots.sh.
> You will also need to update the job card to include "mesoscale".
> For example, #PBS -N jevs_sref_g2o_stats changes to #PBS -N
> jevs_mesoscale_sref_g2o_stats.
> This PR is a good opportunity to take care of that maintenance.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#issuecomment-1646210003>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFMIHXSZVEHONOZ7F23XRLRRVANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA commented:
Yes,

Thanks!

Binbin

On Fri, Jul 21, 2023 at 4:53 PM Shelley Melchior ***@***.***>
wrote:

> And as you are doing in the NARRE PR, remove this line from your driver
> scripts:
> export MET_bin_exec=bin
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#issuecomment-1646226021>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFK4G5RNRTDTLWWAJCDXRLT37ANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented on sorc/compile_evs_global_ens_sorc.sh:
```diff
@@ -4,7 +4,7 @@ SORCevs_gens=`pwd`
 source ../versions/build.ver
 
-source ../modulefiles/v1.0
+source ../modulefiles/v1.0.old
```
This file does not exist in this feature branch. Let's remove the .old extension.

BinbinZhou-NOAA commented:
Sure,

Thanks!

Binbin

On Fri, Jul 21, 2023 at 5:10 PM Shelley Melchior ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In sorc/compile_evs_global_ens_sorc.sh
> <https://github.com/NOAA-EMC/EVS/pull/153#discussion_r1271113594>:
>
> > @@ -4,7 +4,7 @@ SORCevs_gens=`pwd`
>
>  source ../versions/build.ver
>
> -source ../modulefiles/v1.0
> +source ../modulefiles/v1.0.old
>
> This file does not exist in this feature branch. Let's remove the .old
> extension.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#pullrequestreview-1541853629>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFKDSJJQLYSS3AULQGLXRLV33ANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
@BinbinZhou-NOAA , re: COMOUT, it looks like @LoganDawson-NOAA stops his COMOUT construct at $RUN also. I think the directory gets built onto in downstream processing to append the `.$VDATE`. Perhaps the `mkdir -p` needs to be moved to that part of the processing. You are right, having the `mkdir -p` in the J-job will make an empty $RUN directory.

BinbinZhou-NOAA commented:
Shelley,

  So I'd like to set up a "if block":
    if [ $MODELNAME = sref ] ; then
       mkdir $COMOUT.$VDATE
    else
        mkdir $COMOUT
    if

  For sref, mkdir .../atmos.$VDATE
  The other models  keep  .../atmos

Binbin


On Fri, Jul 21, 2023 at 5:49 PM Shelley Melchior ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> , re: COMOUT, it
> looks like @LoganDawson-NOAA <https://github.com/LoganDawson-NOAA> stops
> his COMOUT construct at $RUN also. I think the directory gets built onto in
> downstream processing to append the .$VDATE. Perhaps the mkdir -p needs
> to be moved to that part of the processing. You are right, having the mkdir
> -p in the J-job will make an empty $RUN directory.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#issuecomment-1646270638>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFPMUFBAXDS5CZMIUDDXRL2NTANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
update sorc/compile_evs_global_ens_sorc.sh

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref.ve

BinbinZhou-NOAA committed:
change driver file names, update JEVS_MESOSCALE_PLOTS

BinbinZhou-NOAA committed:
chnage driver file names

BinbinZhou-NOAA committed:
delete useless files

ShelleyMelchior-NOAA commented:
These updates are looking good. Can you also update `modulefiles/mesoscale/mesoscale_stats.sh` the way that you did for plots? Look at the auth EVS repo version and make sure yours includes what is in that one.

BinbinZhou-NOAA committed:
update  modulefiles/mesoscale/mesoscale_stats.sh

BinbinZhou-NOAA commented:
/mesoscale/mesoscale_stats.sh has been updated to match the authorized
version

Binbin

On Sun, Jul 23, 2023 at 2:14 PM Shelley Melchior ***@***.***>
wrote:

> These updates are looking good. Can you also update
> modulefiles/mesoscale/mesoscale_stats.sh the way that you did for plots?
> Look at the auth EVS repo version and make sure yours includes what is in
> that one.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#issuecomment-1646920493>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFNCGICVAUINNWKJQPTXRVSXDANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
update /modulefiles/v1.0 file

ShelleyMelchior-NOAA commented:
Thanks @BinbinZhou-NOAA . I will begin making runs w/ this code in its current state. I still want to revisit the J-jobs to make sure the other developers have a chance to indicate that the STATS and PLOTS will still run correctly for their processing/components.

ShelleyMelchior-NOAA commented:
Found another detail. Update line 3 of `jevs_mesoscale_sref_grid2obs_stats.sh` from
`#PBS -N jevs_mesoscale_sref_g2o_stats`
to
`#PBS -N jevs_mesoscale_sref_grid2obs_stats`

ShelleyMelchior-NOAA commented:
Hi @BinbinZhou-NOAA , I have run the stats processing. Please review the output.
COMOUT= `/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale`
log files:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_cnv_stats.o72906526`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_g2o_stats.o72906547`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_precip_stats.o72906559`

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref.ve

BinbinZhou-NOAA committed:
update driver script files

ShelleyMelchior-NOAA commented on dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_sref_precip_past90days_plots.sh:
```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+#PBS -N jevs_mesoscale_sref_porecip_past90days_plots
```
I see a small typo in line 3. "porecip" should be "precip".

BinbinZhou-NOAA commented:
Shelley,

  The driver script file jevs_mesoscale_sref_grid2obs_stats.sh has been
updated
The other 4 plot driver file have also been modified to match the file name
and job output file name

 For output files:
 The stat files are good, but the subdirectory
/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale/sref.20230722/apcp24mean
is empty. There should be 3 nc files inside.
I'll debug  this issue.

Binbin




On Mon, Jul 24, 2023 at 4:15 PM Shelley Melchior ***@***.***>
wrote:

> Hi @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> , I have run the
> stats processing. Please review the output.
> COMOUT=
> /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale
> log files:
>
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_cnv_stats.o72906526
>
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_g2o_stats.o72906547
>
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_precip_stats.o72906559
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#issuecomment-1648542267>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFIMSWNWIH6IWDW2MTTXR3JU7ANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
Correct a typo in jevs_mesoscale_sref_precip_past90days_plots.sh

BinbinZhou-NOAA commented:
The typo is corrected

Binbin

On Mon, Jul 24, 2023 at 4:20 PM Shelley Melchior ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In
> dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_sref_precip_past90days_plots.sh
> <https://github.com/NOAA-EMC/EVS/pull/153#discussion_r1272714776>:
>
> > @@ -0,0 +1,48 @@
> +#!/bin/bash
> +
> +#PBS -N jevs_mesoscale_sref_porecip_past90days_plots
>
> I see a small typo in line 3. "porecip" should be "precip".
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#pullrequestreview-1544206790>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFNHXA4XONLERWNZVHDXR3KKBANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
update sorc/compile_evs_global_ens_sorc.sh

BinbinZhou-NOAA commented:
Shelley,

  I found the issue:
there is no exec
subdirectory /lfs/h2/emc/vpppg/noscrub/shelley.melchior/githubwkspc/forks/BZ/EVS/exec
which contains some executable files to process SREF's precip member files
 Please go to EVS/sorc and run the script compile_evs_global_ens_sorc.sh
(sref shares this file)
Before doing this, please update this file (I just modified this file again)

Binbin


On Mon, Jul 24, 2023 at 4:32 PM Binbin Zhou - NOAA Affiliate <
***@***.***> wrote:

> Shelley,
>
>   The driver script file jevs_mesoscale_sref_grid2obs_stats.sh has been
> updated
> The other 4 plot driver file have also been modified to match the file
> name and job output file name
>
>  For output files:
>  The stat files are good, but the subdirectory
>
> /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale/sref.20230722/apcp24mean
> is empty. There should be 3 nc files inside.
> I'll debug  this issue.
>
> Binbin
>
>
>
>
> On Mon, Jul 24, 2023 at 4:15 PM Shelley Melchior ***@***.***>
> wrote:
>
>> Hi @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> , I have run
>> the stats processing. Please review the output.
>> COMOUT=
>> /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale
>> log files:
>>
>> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_cnv_stats.o72906526
>>
>> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_g2o_stats.o72906547
>>
>> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_precip_stats.o72906559
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/NOAA-EMC/EVS/pull/153#issuecomment-1648542267>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AQBMPFIMSWNWIH6IWDW2MTTXR3JU7ANCNFSM6AAAAAAZ7E2XPQ>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
>
> --
>
> Binbin Zhou
>
> Physical Scientist
>
> Lynker at NOAA/NWS/NCEP/EMC
>
> 5830 University Research Ct.
>
> College Park, MD 20740
>
> ***@***.***
>
> 301-683-3683
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
Aha! You are correct. I never ran the compile script to generate the executables! I will follow up and re-run the precip stats processing.

ShelleyMelchior-NOAA commented:
Hi @BinbinZhou-NOAA , I updated my clone of your fork's feature branch and then I ran the script in sorc/ that generates the executable files. I re-ran the stats processing for precip and I see the *.nc files in the apcp24mean/ directory now.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_precip_stats.o72916290`

How does that look?

BinbinZhou-NOAA commented:
Shelley,

  The 3 nc files in the directory
/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale/sref.20230722/apcp24mean
are good. The 2 stat files are good as well.

Thanks!

Binbin


On Mon, Jul 24, 2023 at 7:43 PM Shelley Melchior ***@***.***>
wrote:

> Hi @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> , I updated my
> clone of your fork's feature branch and then I ran the script in sorc/ that
> generates the executable files. I re-ran the stats processing for precip
> and I see the *.nc files in the apcp24mean/ directory now.
> COMOUT=/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale
> logfile:
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_precip_stats.o72916290
>
> How does that look?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#issuecomment-1648766182>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFOE3L5LTKEIVI7MHRTXR4CDFANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
HI @BinbinZhou-NOAA , I ran the 4 sref plots jobs. Please review.
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/mesoscale/atmos.20230723`
I untarred the spatial map tar file and the png graphic names look good. I did not view any (xming doesn't work on my windows GFE).
logfiles:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_precip_spatial_plots.o72968908`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_cnv_past90days_plots.o72968890`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_precip_past90days_plots.o72968907`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_grid2obs_past90days_plots.o72968892`

BinbinZhou-NOAA commented:
Shelley,

  I have loaded the graphic files to the SREF web page from your runs to
review:
https://www.emc.ncep.noaa.gov/users/verification/binbin.zhou/sref
The plots are all looking good!

Thanks!

Binbin

On Tue, Jul 25, 2023 at 10:54 AM Shelley Melchior ***@***.***>
wrote:

> HI @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> , I ran the 4
> sref plots jobs. Please review.
> COMOUT=
> /lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/mesoscale/atmos.20230723
> I untarred the spatial map tar file and the png graphic names look good. I
> did not view any (xming doesn't work on my windows GFE).
> logfiles:
>
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_precip_spatial_plots.o72968908
>
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_cnv_past90days_plots.o72968890
>
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_precip_past90days_plots.o72968907
>
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_sref_grid2obs_past90days_plots.o72968892
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#issuecomment-1649998593>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFMI2ANXDPLZZF5ZGWDXR7M2JANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
Great! This PR is nearly complete, @BinbinZhou-NOAA  I want to touch on 2 more things.
1 - please remove line 96 from `JEVS_MESOSCALE_STATS`.
2 - for the shared J-job files, I want other developers to take a peek at `JEVS_MESOSCALE_PLOTS` and `JEVS_MESOSCALE_STATS` to make sure the files will work as intended for other models in the mesoscale component. @malloryprow , @RoshanShrestha-NOAA , @MarcelCaron-NOAA

malloryprow commented:
@ShelleyMelchior-NOAA I think that JEVS_MESOSCALE_STATS looks good. I had caught some things a few weeks ago that might have broken things, but these were resolved.

MarcelCaron-NOAA commented:
> Great! This PR is nearly complete, @BinbinZhou-NOAA I want to touch on 2 more things. 1 - please remove line 96 from `JEVS_MESOSCALE_STATS`. 2 - for the shared J-job files, I want other developers to take a peek at `JEVS_MESOSCALE_PLOTS` and `JEVS_MESOSCALE_STATS` to make sure the files will work as intended for other models in the mesoscale component. @malloryprow , @RoshanShrestha-NOAA , @MarcelCaron-NOAA

@ShelleyMelchior-NOAA Changes to JEVS_MESOSCALE_PLOTS won't interfere with other models in the mesoscale component as far as I can tell.

RoshanShrestha-NOAA commented:
> Great! This PR is nearly complete, @BinbinZhou-NOAA I want to touch on 2 more things.
> 1 - please remove line 96 from `JEVS_MESOSCALE_STATS`.
> 2 - for the shared J-job files, I want other developers to take a peek at `JEVS_MESOSCALE_PLOTS` and `JEVS_MESOSCALE_STATS` to make sure the files will work as intended for other models in the mesoscale component. @malloryprow , @RoshanShrestha-NOAA , @MarcelCaron-NOAA

The JEVS_MESOSCALE_STATS looks good to me too.

BinbinZhou-NOAA committed:
update JEVS_MESOSCALE_STATS by removing line 96

ShelleyMelchior-NOAA commented:
Thanks all! @BinbinZhou-NOAA , almost there. Can you update your feature branch to bring in the latest from auth EVS develop?

BinbinZhou-NOAA commented:
Shelley,

 The line 96 in JEVS_MESOSCALE_STATS has been removed

Binbin

On Tue, Jul 25, 2023 at 12:39 PM Shelley Melchior ***@***.***>
wrote:

> Great! This PR is nearly complete, @BinbinZhou-NOAA
> <https://github.com/BinbinZhou-NOAA> I want to touch on 2 more things.
> 1 - please remove line 96 from JEVS_MESOSCALE_STATS.
> 2 - for the shared J-job files, I want other developers to take a peek at
> JEVS_MESOSCALE_PLOTS and JEVS_MESOSCALE_STATS to make sure the files will
> work as intended for other models in the mesoscale component. @malloryprow
> <https://github.com/malloryprow> , @RoshanShrestha-NOAA
> <https://github.com/RoshanShrestha-NOAA> , @MarcelCaron-NOAA
> <https://github.com/MarcelCaron-NOAA>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#issuecomment-1650176030>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFO7CRP644EEUG4Q2LLXR7ZDJANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref.ve

BinbinZhou-NOAA commented:
Yes, done!

Binbin

On Tue, Jul 25, 2023 at 2:18 PM Shelley Melchior ***@***.***>
wrote:

> Thanks all! @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> ,
> almost there. Can you update your feature branch to bring in the latest
> from auth EVS develop?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/153#issuecomment-1650319444>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFN343ERPM64TKHG27TXSAEYFANCNFSM6AAAAAAZ7E2XPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
Excellent! I have been doing the code review as we have gone along. I will formalize the code review and then merge this PR.

ShelleyMelchior-NOAA reviewed (approved):
These changes are tested and approved.
